### DISCUSSION TITLE
How to get the Windows toast notification?

### DISCUSSION THREAD

**Discussion by wsw70** [ncruces/zenity#97] [Q&A] [answered]
I just tried on a Windows 11 to raise a [toast notification](https://github.com/ncruces/zenity/wiki/Notification#windows).

Unfortunately, I only get the one from Windows Server Core - they are noticeably different because the Windows Server Core one requires an acknowledgment from the user, while the toast does not.

**Comment by ncruces** [comment, marked as answer]
The one from Server Core is the **third** fallback [here](https://github.com/ncruces/zenity/blob/master/notify_windows.go):
1. is PowerShell, running this [script](https://github.com/ncruces/zenity/blob/master/notify_windows.go#L176-L205).
2. is [`Shell_NotifyIconW`](https://learn.microsoft.com/en-us/windows/win32/api/shellapi/nf-shellapi-shell_notifyiconw), which should work on any desktop OS.
3. is [`WTSSendMessageW`](https://learn.microsoft.com/en-us/windows/win32/api/wtsapi32/nf-wtsapi32-wtssendmessagew), which works even if you don't have [Explorer](https://en.wikipedia.org/wiki/File_Explorer).

TBH I'm not sure what's going on, how those could fail.

**Reply by wsw70** [reply]
OK I know. I was doing my tests from the vscode integrated terminal - I guess that it is not correctly recognized.
I tried to run a built exec from Windiws terminal (a powershell prompt) and it works as expected.

Sorry for the disturbance.

EDIT: for some reason now it works fine even in vscode. Go figure :)
